static/js/server_events_dispatch.js

```diff
@@ -348,6 +348,21 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
         ui.remove_message(msg_id);
         break;
 
+    case 'attachments':
+        if (event.op === 'add') {
+            page_params.attachments.push(event.attachment);
+        } else if (event.op === 'remove') {
+            page_params.attachments = page_params.attachments.filter(function (a) {
+                return a.path_id !== event.attachment.path_id;
+            });
+        }
+        // re-render list only when the attachments-settings
+        // page is loaded
+        if ($("#attachments-settings").length) {
+            attachments_ui.set_up_attachments();
```

Comment:
this seems like overkill for adding/removing a single attachment...

Reply:
@timabbott what will be a better way?

Reply:
I think @timabbott meant to do something like this. 

```
page_params.attachments = event.attachments
```
You can get rid of `add` and `remove` events and use a similar approach as that of `default_streams`.

Reply:
```js
// re-render list only when the attachments-settings
// page is loaded
    if ($("#attachments-settings").length) {
         attachments_ui.set_up_attachments();
   }
```
@AasthaGupta Can you remove this. This was what Tim was referring to which I misinterpreted badly.

Reply:
@hackerkid If I remove this line, then the attachments list in settings in not re-rendered and hence not updated in real time.

Reply:
Right, the idea is that you should be able to have this just update the `list_render` data set, rather than doing a full rerender.